### DISCUSSION TITLE
is it possible to generate multiple cucumber json file

### DISCUSSION THREAD

**Discussion by peacetrader** [badeball/cypress-cucumber-preprocessor#1177] [Q&A] [answered]
Hi, how to generate multiple cucumber json file for each run when we do parallel run?.
currently in package.json we specify
**"cypress-cucumber-preprocessor": {
"json": {
"enabled": true,
"output": "report/cucumber.json"
}
}**
When we run the tests parallely, the cucumber.json gets generated for the last execution, so the previous ones are getting override. I am using cypress-parallel where I specify the no of threads in the command. This will generate one cucumber.json file. Would it be possible to generate multiple json files? Would you be able to help?.
This can help everyone who uses cucumber and multiple-html-cucumber-report for parallel run.
commands : 
cypress-parallel -s cy:run -t 2 -a "cypress command args"
cy:run : "cypress run --browser edge"
This command will split the feature files and run them in 2 threads.

**Comment by badeball** [comment, marked as answer]
You can use [`process.env.CYPRESS_THREAD`](https://github.com/tnicola/cypress-parallel?tab=readme-ov-file#cypress_thread) to detect parallelization and configure a [dynamic outpath path](https://github.com/badeball/cypress-cucumber-preprocessor/issues/1055#issuecomment-1631337744).

**Reply by peacetrader** [reply]
Thank you. do we have to pass jsonOutput as an env variable in order to use this approach?

**Reply by badeball** [reply]
No, you configure it during `setupNodeEvents`, as shown in the last link above.
